python/ray/train/lightning/_lightning_utils.py

```diff
@@ -270,7 +270,7 @@ def on_train_epoch_end(self, trainer, pl_module) -> None:
         train.report(metrics=metrics, checkpoint=checkpoint)
 
         # Add a barrier to ensure all workers finished reporting here
-        torch.distributed.barrier()
+        trainer.strategy.barrier()
```

Comment:
Any way we can test this?

Reply:
The error is non-deterministic and never happened before. I know it should be this but I am not able to reproduce an error here. What about having a stress test that saves checkpoint in a high frequency and ensure it never hangs?